main.go

```diff
@@ -98,6 +99,7 @@ func main() {
 			for {
 				prs, resp, _ := client.Search.Issues(
 					// Search query to find PR's
+					context.Background(),
```

Comment:
> How can I identify that passing c is working as expected? It doesnt give me any errors and starts up my server. Is that enough of an indicator?

Yep, then it should be fine.

> Honestly I have not really understood what passing the context to go-github does.

In your situation, practically, it probably does not have much effect. One of the ways it can help is if the incoming HTTP request to your server is canceled early (imagine, the user closes their browser window while the page was loading), then the relevant request to github servers will also be cancelled automatically and avoid doing extra unnecessary work.

However, from a quick look at the current version of gin, it [doesn't seem they support request cancellation](https://github.com/gin-gonic/gin/blob/e2212d40c62a98b388a5eb48ecbdcf88534688ba/context.go#L584-L586).

In the general case, it's a good idea to propagate context for the reasons described in https://blog.golang.org/context. But it's more helpful and applicable to larger projects where you're doing tracing, or if you want to set specific timeouts or deadlines, etc.